mlflow/pyfunc/__init__.py

```diff
@@ -716,6 +719,64 @@ def load_model(
     return PyFuncModel(model_meta=model_meta, model_impl=model_impl)
 
 
+def _download_model_conda_env(model_uri):
+    model_meta_path = _download_artifact_from_uri(os.path.join(model_uri, MLMODEL_FILE_NAME))
+    model_meta = Model.load(model_meta_path)
+    conda_yml_file_name = model_meta.flavors[FLAVOR_NAME][ENV]
+    return _download_artifact_from_uri(os.path.join(model_uri, conda_yml_file_name))
+
+
+def get_model_dependencies(model_uri, format="pip"):
+    """
+    Given a model URL, and format, return the downloaded dependency file path.
+    Available dependency file format includes "pip" and "conda".
+    """
+    req_file_uri = os.path.join(model_uri, _REQUIREMENTS_FILE_NAME)
+
+    if format == "pip":
+        try:
+            pip_file_path = _download_artifact_from_uri(req_file_uri)
+        except Exception as e:
+            # fallback to download conda.yaml file and parse the "pip" section from it.
```

Comment:
Can we print (`_logger.info()`) a message in this case telling the user that we're going to fetch pip requirements from the conda.yaml file and that other conda dependencies will be ignored?

Reply:
In general, if the user specifies `format=pip`, I think it would be useful to download the `conda.yaml` and check if there are non-pip requirements. If there are non-pip requirements in `conda.yaml`, we should print a warning to the user indicating that these requirements will be excluded.